raiden/storage/migrations/v19_to_v20.py

```diff
@@ -0,0 +1,168 @@
+            token_network_identifier = state_change_data['token_network_identifier']
+            channel_identifier = state_change_data['channel_identifier']
+
+            channel_new_state_change = _find_channel_new_state_change(
+                storage=storage,
+                token_network_address=token_network_identifier,
+                channel_identifier=channel_identifier,
+            )
+
+            if not channel_new_state_change.data:
+                raise RaidenUnrecoverableError(
+                    'Could not find the state change for channel {channel_identifier}, '
+                    'token network address: {token_network_identifier} being created. ',
+                )
+
+            channel_state_data = json.loads(channel_new_state_change.data)
+            new_channel_state = channel_state_data['channel_state']
+            our_locksroot, partner_locksroot = get_onchain_locksroots(
+                raiden=raiden,
+                token_network_address=token_network_identifier,
+                channel_identifier=channel_identifier,
+                participant1=new_channel_state['our_state']['address'],
+                participant2=new_channel_state['partner_state']['address'],
+                block_identifier='latest',
+            )
+
+            data['our_onchain_locksroot'] = our_locksroot
+            data['partner_onchain_locksroot'] = partner_locksroot
+
+            updated_state_changes.append((
+                json.dumps(data),
+                state_change.state_change_identifier,
+            ))
+        storage.update_state_changes(updated_state_changes)
+
+
+def _add_onchain_locksroot_to_snapshot(
+        raiden: RaidenService,
+        storage: SQLiteStorage,
+        raw_snapshot: Dict[str, Any],
+):
+    """
+    Add `onchain_locksroot` to each NettingChannelEndState
+    """
+    snapshot = json.loads(raw_snapshot)
+
+    for payment_network in snapshot.get('identifiers_to_paymentnetworks', {}).values():
+        for token_network in payment_network.get('tokennetworks', []):
+            channelidentifiers_to_channels = token_network.get(
+                'channelidentifiers_to_channels',
+                dict(),
+            )
+            for channel in channelidentifiers_to_channels.values():
+                our_locksroot, partner_locksroot = _get_onchain_locksroots(
+                    raiden=raiden,
+                    storage=storage,
+                    token_network=token_network,
+                    channel=channel,
+                )
+
+                channel['our_state']['onchain_locksroot'] = our_locksroot
+                channel['partner_state']['onchain_locksroot'] = partner_locksroot
+
+    return json.dumps(snapshot, indent=4)
+
+
+def _add_onchain_locksroot_to_snapshots(
+        raiden: RaidenService,
+        storage: SQLiteStorage,
+):
+    updated_snapshots_data = []
+    for snapshot in storage.get_snapshots():
```

Comment:
You can parallelize this with a gevent Pool as we do [here](https://github.com/raiden-network/raiden/blob/2b2b205f2c2bec64ed74b06dc2aeb7aa0503dbb6/raiden/storage/migrations/v18_to_v19.py#L181-L201).